demonstrations/tutorial_liealgebra.py

```diff
@@ -0,0 +1,392 @@
+Note also that we are talking about a **vector** space in the mathematical sense, and the elements ("vectors") in :math:`\mathfrak{g}` are actually operators (matrices) in our case looking at quantum physics.
+
+One very relevant Lie algebra for us is the special unitary algebra :math:`\mathfrak{su}(N)`, the space of :math:`N \times N` skew-Hermitian matrices with trace zero.
+The fact that we look at skew-Hermitian (:math:`H^\dagger = - H`) instead of Hermitian (:math:`H^\dagger = H`) matrices is a technical detail (see note below). For all practical purposes you
+can just think of Hermitian operators with an imaginary factor and note that linear cominations are strictly over the reals. In fact, you may sometimes
+find references to :math:`\mathfrak{su}(N)` being the Hermitian matrices in physics literature (see `wikipedia <https://en.wikipedia.org/wiki/Special_unitary_group#Fundamental_representation>`_).
+
+.. note::
+
+    The result of a commutator between two Hermitian operators :math:`H_1` and :math:`H_2` is always skew-Hermitian due to the commutator's anti-commutativity, i.e.
+
+    .. math:: [H_1, H_2]^\dagger = [H_2^\dagger, H_1^\dagger] = - [H_1, H_2].
+
+    This means that Hermitian operators are not closed under commutation, and thus do not form a Lie algebra
+    (because the commutator maps outside the set of Hermitian matrices). But instead, skew-Hermitian operators do.
+    Note that the algebra of :math:`N \times N` skew-Hermitian matrices is called the unitary algebra :math:`\mathfrak{u}(N)`, whereas
+    the additional property of the trace being zero making it the `special` unitary algebra :math:`\mathfrak{su}(N)`. They generate
+    the unitary group :math:`U(N)` and the special unitary group :math:`SU(N)` with determinant 1, respectively.
+
+
+The Pauli matrices :math:`\{iX, iY, iZ\}` span the :math:`\mathfrak{su}(2)` algebra that we can associate with single qubit dynamics.
+For multiple qubits we have 
+
+.. math:: \mathfrak{su}(2^n) = \text{span}_{\mathbb{R}}\left(\{iX_0, .., iY_0, .., iZ_0, .., iX_0 X_1, .. iY_0 Y_1, .., iZ_0 Z_1, ..\}\right),
+
+where the span is over the reals :math:`\mathbb{R}`. In particular, we cannot do a complex span over Paulis, since this could destroy the anti-commutativity again. Another way
+of thinking about this is that Lie algebra elements "live" in the exponent of a unitary operator, and having that exponent become Hermitian instead of skew-Hermitian
+destroys the unitary property.
+
+Let us briefly test some of these properties numerically.
+First, let us do a linear combination of :math:`\{iX, iY, iZ\}` with some real values and check unitarity after putting them in the exponent.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane import X, Y, Z
+qml.operation.enable_new_opmath()
+
+su2 = [1j * X(0), 1j * Y(0), 1j * Z(0)]
+
+coeffs = [1., 2., 3.]
+exponent = sum([c * P for c,P in zip(coeffs, su2)])
```

Comment:
Not an important suggestion but `qml.dot` works here as well:
```suggestion
exponent = qml.dot(coeffs, su2)
```